core/src/execution/datafusion/expressions/negative.rs

```diff
@@ -0,0 +1,304 @@
+                            array,
+                            arrow::array::Int8Array,
+                            i8::MIN,
+                            i8::MAX,
+                            "integer"
+                        ),
+                        DataType::Int16 => check_overflow!(
+                            array,
+                            arrow::array::Int16Array,
+                            i16::MIN,
+                            i16::MAX,
+                            "integer"
+                        ),
+                        DataType::Int32 => check_overflow!(
+                            array,
+                            arrow::array::Int32Array,
+                            i32::MIN,
+                            i32::MAX,
+                            "integer"
+                        ),
+                        DataType::Int64 => check_overflow!(
+                            array,
+                            arrow::array::Int64Array,
+                            i64::MIN,
+                            i64::MAX,
+                            "integer"
+                        ),
+                        DataType::Float32 => check_overflow!(
+                            array,
+                            arrow::array::Float32Array,
+                            f32::MIN,
+                            f32::MAX,
+                            "float"
+                        ),
+                        DataType::Float64 => check_overflow!(
+                            array,
+                            arrow::array::Float64Array,
+                            f64::MIN,
+                            f64::MAX,
+                            "float"
+                        ),
+                        DataType::Interval(value) => match value {
+                            arrow::datatypes::IntervalUnit::YearMonth => check_overflow!(
+                                array,
+                                arrow::array::IntervalYearMonthArray,
+                                i32::MIN,
+                                i32::MAX,
+                                "interval"
+                            ),
+                            arrow::datatypes::IntervalUnit::DayTime => check_overflow!(
+                                array,
+                                arrow::array::IntervalDayTimeArray,
+                                i64::MIN,
+                                i64::MAX,
+                                "interval"
+                            ),
+                            arrow::datatypes::IntervalUnit::MonthDayNano => check_overflow!(
+                                array,
+                                arrow::array::IntervalMonthDayNanoArray,
+                                i128::MIN,
+                                i128::MAX,
+                                "interval"
+                            ),
+                        },
+                        _ => unimplemented!(
```

Comment:
So if other types come in like decimal, this will fail?

Reply:
Hmm, i have change it to not raise a panic.

There is no overflow checks for decimals for unary minus in spark.